python/tank/util/defaults_manager.py

```diff
@@ -24,6 +24,6 @@ def get_http_proxy(self):
     def get_user(self):
         from . import shotgun
         data = shotgun.get_associated_sg_config_data()
-        if "api_script" in data:
+        if "api_script" in data and data["api_script"]:
```

Comment:
Yes, definitely need to both tweak this code and add comments - but I didn't want to start messing with the security related APIs and stuff since I know you are on the case already -- too many chefs in the kitchen! So I leave the security API related stuff to you to tweak, refactor and improve - I just added the minimal bits needed in order to get the shotgun integration working. Hope that makes sense!

In terms of code review, I have quite a few comments around the code in `get_user`, but because it still constructs the User objects directly, I know it is about to change, so I thought there is no point starting to review/clean up at this point. Better to leave that until we feel it's actually ready. Hope it makes sense!